src/widget/sidebar/extra.c

```diff
@@ -385,44 +394,100 @@ static int draw_request_buttons(int y_offset)
-                text_draw_number(r->available, 0, ")", width, y_offset, FONT_NORMAL_WHITE);
+                int is_stockpiled = city_resource_is_stockpiled(r->resource);
+                int enough_resource = 0;
+
+                int request_index = data.requests[i].index;
+                if (city_request_has_troop_request() && i != 0) {
+                    request_index++;
+                }
+
+                int status = city_request_get_status(request_index);
+
+                // button text
+                if (status) {
+                    if (status == CITY_REQUEST_STATUS_NOT_ENOUGH_RESOURCES) {
+                        if (is_stockpiled) {
+                            image_draw(assets_get_image_id(assets_get_group_id("Areldir", "UI_Elements"),
+                                                           "Store Icon"), data.x_offset + 5, y_offset + 10);
+                            text_draw_centered(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_UNSTOCK),
+                                               data.x_offset + 2 , y_offset + 25, 158, FONT_NORMAL_GREEN, 0);
+                        } else {
+                            text_draw_centered(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_STOCK),
+                                               data.x_offset + 2 , y_offset + 25, 158, FONT_NORMAL_GREEN, 0);
+                        }
+                    } else {
+                        enough_resource = 1;
+                        text_draw_centered(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_SEND),
+                                           data.x_offset + 2 , y_offset + 25, 158, FONT_NORMAL_GREEN, 0);
+                    }
+                }
+
+                // request current / total
+                width += text_draw_number(r->available, 0, "/", width, y_offset+2, enough_resource ? FONT_NORMAL_GREEN : FONT_NORMAL_RED);
+                width += text_draw_number(r->amount, 0, "",
+                                          width-5, y_offset+2, enough_resource ? FONT_NORMAL_GREEN : FONT_NORMAL_RED);
+
+            } else {
+                width += text_draw_number(r->amount, 0, "",
+                                          width, y_offset+2, FONT_NORMAL_GREEN);
+
+                text_draw_centered(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_SEND),
+                                   data.x_offset + 2 , y_offset + 25, 158, FONT_NORMAL_GREEN, 0);
             }
-            y_offset += EXTRA_INFO_LINE_SPACE;
-            lang_text_draw_amount(8, 4, r->time, data.x_offset + 4, y_offset + 4,
-                r->time <= REQUEST_MONTHS_LEFT_FOR_RED_WARNING ? FONT_NORMAL_RED : FONT_NORMAL_WHITE);
+
+            // request time left
+            width += text_draw_number(r->time, 0, "", width, y_offset+2,
+                                      r->time <= REQUEST_MONTHS_LEFT_FOR_RED_WARNING ? FONT_NORMAL_RED : FONT_NORMAL_GREEN);
+            text_draw_ellipsized(translation_for(TR_SIDEBAR_EXTRA_REQUESTS_MONTHS), width - 2, y_offset+2,
```

Comment:
You could use `lang_text_draw_ellipsized(8, 4 + (r->time != 1), width - 2, y_offset+2,` instead, which will use the built-in strings `Month` or `Months` depending on the number of months remaining.

Reply:
yes I will fix it